pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/patch/DeletePatch.java

```diff
@@ -72,9 +75,25 @@ Optional<Object> processSingleNameValue() {
         return result;
     }
 
-    //TODO this is supposed to get handled by PEPPER-1209
-    private boolean isSafeToDelete() {
-        return true;
+    public static boolean isSafeToDelete(List<Integer> tissueIds) {
+        if (tissueIds == null || tissueIds.isEmpty())
+            return true;
+        return clinicalOrderDao.getClinicalOrdersForTissueIds(tissueIds).values().stream().filter(list -> !list.isEmpty()).findAny()
+                .isEmpty();
+    }
+
+    public static boolean isSafeToDelete(Patch patch) {
```

Comment:
Consider this non-static implementation instead:

```
public boolean isSafeToDelete() {
        List<Integer> tissueIds = new ArrayList<>();
        if (Patch.isTissueDeletePatch(patch)) {
            tissueIds.add(Integer.parseInt(patch.getId()));
        } else if (Patch.isSmIdDeletePatch(patch)) {
            int tissueId = new TissueSMIDDao().get(Integer.parseInt(patch.getId())).getTissueId();
            tissueIds.add(tissueId);
        } else if (Patch.isOncHistoryDeletePatch(patch)) {
            tissueIds = TissueDao.getTissuesByOncHistoryDetailId(Integer.parseInt(patch.getId()));
        }
        return isSafeToDelete(tissueIds);
    }
```

Reply:
Similarly:

```
public boolean isSafeToDelete(List<Integer> tissueIds) {
        if (tissueIds == null || tissueIds.isEmpty()) {
            return true;
        }
        return new ClinicalOrderDao().getClinicalOrdersForTissueIds(tissueIds).values().stream().filter(list -> !list.isEmpty()).findAny()
                .isEmpty();
    }
```

Reply:
Is there any concerns to make a ClinicalOrder globally for the class and call it here? I want to make sure we avoid any unnecessary memory consumtions

Reply:
I'm not sure what you mean by "globally for the class".  Do you mean an instance variable?  If so, that should be fine IMO.  The memory use for a field is vanishingly small compared to all the other operations.